cnxarchive/utils/ident_hash.py

```diff
@@ -82,3 +81,186 @@ def join_ident_hash(id, version):
     if version:
         join_args.append(version)
     return HASH_CHAR.join(join_args)
+
+
+class CNXHash(uuid.UUID):
```

Comment:
I find it kind of strange that we're subclassing `uuid.UUID` class.  What is the reason for this?